lua/lspconfig/server_configurations/foam.lua

```diff
@@ -0,0 +1,26 @@
+local util = require 'lspconfig.util'
+
+return {
+  default_config = {
+    -- `foam-ls` can be a shell script executing:
+    -- node /path/to/foam-language-server/lib/foam-ls.js --stdio
+    cmd = { 'foam-ls' },
```

Comment:
Wouldn't it be better if the npm package provided an executable itself? That way when someone installs it you'll be able to have a common, canonical, name for the executable that everyone can use.

Reply:
Yes it sure provides one; [package.json#L24](https://github.com/FoamScience/foam-language-server/blob/fe6aa2e5cf9c37c15fc60d0b1164958397eff15f/package.json#L24)
Should I remove the comment?

Reply:
For npm packages, the following will also be needed (you can refer to other npm-based configs such as `tsserver.lua`):

```lua
local bin_name = 'foam-ls'
local cmd = { 'foam-ls' }

if vim.fn.has 'win32' == 1 then
  cmd = { 'cmd.exe', '/C', bin_name }
end
```